File: dotnet/DemoApp/Filters/FunctionInvocationRetryFilter.cs
using Microsoft.SemanticKernel;

namespace Filters;

internal sealed class FunctionInvocationRetryFilter(ushort times = 3)
    : IFunctionInvocationFilter
{
    public async Task OnFunctionInvocationAsync(
        FunctionInvocationContext context,
        Func<FunctionInvocationContext, Task> next)
    {
        ushort count = 0;
        do
        {
            try
            {
                await next(context);
            }
            catch
            {
            }
        } while (count++ < times);
    }
}

File: python/DemoApp/core/utilities/filters/functioninvocationretryfilter.py
from semantic_kernel.filters.functions.function_invocation_context import FunctionInvocationContext
from collections.abc import Callable, Coroutine
from typing import Any


class FunctionInvocationRetryFilter:
    def __init__(self, max_retries = 5) -> None:
        self._max_retries = max_retries

    async def OnFunctionInovationAsync(self, context: FunctionInvocationContext,
    next: Callable[[FunctionInvocationContext], Coroutine[Any, Any, None]]) -> None:
        
        count = 0
        while (count < self._max_retries):
            try:
                await next(context)
                break
            except Exception as e:
                count += 1
                print(f"Retrying function {context.function.name} due to error: {e}")
                continue
